pkg/apis/coordination/types.go

```diff
@@ -67,3 +78,45 @@ type LeaseList struct {
 	// items is a list of schema objects.
 	Items []Lease
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LeaseCandidate defines an identity lease concept.
+type LeaseCandidate struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	Spec LeaseCandidateSpec
+}
+
+// LeaseCandidateSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// BinaryVersion is the binary version
+	BinaryVersion string
+	// CompatibilityVersion is the compatibility version
+	CompatibilityVersion string
+	// TargetLease is a name/namespace pair of the lease that the candidate can lead
+	TargetLease string
```

Comment:
these were annotations in the KEP. I would at least try to preprare for other strategies by turning this into an extensible sum type.